src/coorddesc.jl

```diff
@@ -84,15 +84,19 @@ mutable struct CoordinateDescentState{T}
     HHt::Matrix{T}
     XHt::Matrix{T}
     XtW::Matrix{T}
+    WHHt_j::Matrix{T}
+    HtWtW_j::Matrix{T}
     violation::T
     violation_init::Union{Nothing, T}
     
     function CoordinateDescentState{T}(X, W, H, violation, violation_init) where T
         p, n, k = nmf_checksize(X, W, H)
-        new{T}(W * H, 
+        new{T}(W * H,    
```

Comment:
This line has trailing whitespace that would ideally be deleted. See https://bobbyhadz.com/blog/remove-trailing-whitespace-vscode for instructions about doing this automatically in vscode.